modules/querying/pages/query-optimizer/index.adoc

```diff
@@ -0,0 +1,34 @@
+= Query Optimizer
+:description: Overview of the query optimizer.
+
+GSQL has a built-in query optimizer that can optimize the execution plan of a query when you install the query.
+As of TigerGraph 3.7, the optimizer can optimize a query in the following ways:
+
+* <<_cost_based_optimization>>
+
+== What is the query optimizer?
+The query optimizer optimizes the compiled binary execution plan of a query.
+
+When you xref:query-operations.adoc#_install_query[install a GSQL query], GSQL compiles the query logic into a binary execution plan.
+If your query contains a xref:querying:select-statement/index.adoc#_path_pattern[path pattern], the pattern contains a source vertex set, one or more path edge patterns, and one or more step vertex sets.
```

Comment:
Oh I think I misread it. Are you defining what a path pattern is? I thought the sentence was missing a predicate. I find it a little confusing since this is the only reference to path patterns.